### PR TITLE
Feast 0.14 support

### PR DESCRIPTION
**Pull request by Felix-neko**
I've made some changes to support 0.14 ``feast`` API.

Also we're checking some extra settings in Hive (hive sandbox was based on https://hshirodkar.medium.com/apache-hive-on-docker-4d7280ac6f8e tutorial).

### CODE REVIEW

**Comment by itsmesrds** [comment]
@Felix-neko can you pls add example file as well. Thanks

**Comment by Felix-neko** [comment]
> @Felix-neko can you pls add example file as well. Thanks

Yes, of course ^_^

**Comment by itsmesrds** [comment]
```
TypeError                                 Traceback (most recent call last)
/var/folders/n7/zfjw6bs53qj1sq3wzdd_czk40000gn/T/ipykernel_3873/4214990728.py in <module>
     19 print("Retrieving training data...")
     20 
---> 21 training_df = fs.get_historical_features(features=features, entity_df=entity_df).to_df()
     22 
     23 print()

/usr/local/anaconda3/envs/feast_dev/lib/python3.8/site-packages/feast/infra/offline_stores/offline_store.py in to_df(self)
     41     def to_df(self) -> pd.DataFrame:
     42         """Return dataset as Pandas DataFrame synchronously including on demand transforms"""
---> 43         features_df = self._to_df_internal()
     44         if self.on_demand_feature_views is None:
     45             return features_df

/usr/local/anaconda3/envs/feast_dev/lib/python3.8/site-packages/feast_hive/hive.py in _to_df_internal(self)
    304 
    305     def _to_df_internal(self) -> pd.DataFrame:
--> 306         return self.to_arrow().to_pandas()
    307 
    308     def _to_arrow_internal(self) -> pa.Table:

/usr/local/anaconda3/envs/feast_dev/lib/python3.8/site-packages/feast/infra/offline_stores/offline_store.py in to_arrow(self)
     64         """Return dataset as pyarrow Table synchronously"""
     65         if self.on_demand_feature_views is None:
---> 66             return self._to_arrow_internal()
     67 
     68         features_df = self._to_df_internal()

/usr/local/anaconda3/envs/feast_dev/lib/python3.8/site-packages/feast_hive/hive.py in _to_arrow_internal(self)
    307 
    308     def _to_arrow_internal(self) -> pa.Table:
--> 309         with self._queries_generator() as (queries, final_feature_names):
    310             # print(final_feature_names)
    311             with self._conn.cursor() as cursor:

/usr/local/anaconda3/envs/feast_dev/lib/python3.8/contextlib.py in __enter__(self)
    111         del self.args, self.kwds, self.func
    112         try:
--> 113             return next(self.gen)
    114         except StopIteration:
    115             raise RuntimeError("generator didn't yield") from None

/usr/local/anaconda3/envs/feast_dev/lib/python3.8/site-packages/feast_hive/hive.py in query_generator()
    231                     full_feature_names=full_feature_names)
    232 
--> 233                 rendered_query = offline_utils.build_point_in_time_query(
    234                     query_contexts,
    235                     left_table_query_string=table_name,

TypeError: build_point_in_time_query() got an unexpected keyword argument 'entity_df_columns'
```
I'm using feast==0.14.0 and this PR

@Felix-neko Any idea about this ??

**Comment by Felix-neko** [comment]
> @Felix-neko Any idea about this ??

Hmm... Let's look... What did you run to get this error?

P.S. Here's my feast sandbox: https://github.com/Felix-neko/feast_sandbox, i get a couple of test repos and a docker sandbox for Hive there. Maybe it will be useful?

**Comment by itsmesrds** [comment]
@Felix-neko looks like you were using the master feast branch i guess. it's still incompatible with 0.14.0 or 0.14.1

**Comment by Felix-neko** [comment]
> @Felix-neko looks like you were using the master feast branch i guess. it's still incompatible with 0.14.0 or 0.14.1

Hmmm.... And does it work if you use feast from master branch on your machine?

**Comment by itsmesrds** [comment]
> > @Felix-neko looks like you were using the master feast branch i guess. it's still incompatible with 0.14.0 or 0.14.1
> 
> Hmmm.... And does it work if you use feast from master branch on your machine?

@Felix-neko  I'm unable to install the master branch feast (because of no guide). but I looked in to the code, those params are matching. if you are able to share the master branch wheel file, it will be great or steps to install from master branch. 


Thanks

**Comment by Felix-neko** [comment]
Well, try to do the following:

Clone feast repo
pips install grpcio-tools
go to https://github.com/feast-dev/feast/tree/master/sdk/python folder and run `python3 setup install`

**Comment by itsmesrds** [comment]
> pip3 install grpcio-tools

have tried to follow this script 

make package-protos && make compile-protos-python && infra/scripts/publish-python-sdk.sh  --directory-path sdk/python --repository pypi

but, for the custom build, getting 
```
ModuleNotFoundError: No module named 'proto'
```
even after install protobuff , not sure where I'm missing

**Comment by itsmesrds** [comment]
Im able to run the code in this PR without any errors but I'm not getting the data correctly(getting NAN's).  here is my code:

```from datetime import timedelta

from feast import Entity, Feature, FeatureView, ValueType


from feast import Entity, Feature, FeatureView, ValueType
from feast_hive import HiveSource


driver_hourly_stats = HiveSource(
#     table='driver_stats',
    query = """
    SELECT Timestamp(cast(event_timestamp / 1000000 as bigint)) AS event_timestamp,
           driver_id, conv_rate, acc_rate, avg_daily_trips,
           Timestamp(cast(created / 1000000 as bigint)) AS created
    FROM default.driver_stats
    """,
    event_timestamp_column="event_timestamp",
    created_timestamp_column="created",
)

# Define an entity for the driver.
driver = Entity(name="driver_id", value_type=ValueType.INT64, description="driver id", )

# Define FeatureView
driver_hourly_stats_view = FeatureView(
    name="driver_hourly_stats",
    entities=["driver_id"],
    ttl=timedelta(weeks=52),
    features=[
        Feature(name="conv_rate", dtype=ValueType.FLOAT),
        Feature(name="acc_rate", dtype=ValueType.FLOAT),
        Feature(name="avg_daily_trips", dtype=ValueType.INT64),
    ],
    online=True,
    batch_source=driver_hourly_stats,
    tags={},
)
```

```
from datetime import datetime, timedelta

import pandas as pd
from driver_repo import driver, driver_hourly_stats, driver_hourly_stats_view

from feast import FeatureStore



pd.set_option("display.max_columns", None)
pd.set_option("display.width", 1000)


fs = FeatureStore(repo_path=".")

# Deploy the feature store to AWS
print("Deploying feature store to AWS...")
fs.apply([driver, driver_hourly_stats_view])

# Select features
features = ["driver_hourly_stats:conv_rate", "driver_hourly_stats:acc_rate"]

# Create an entity dataframe. This is the dataframe that will be enriched with historical features
entity_df = pd.DataFrame(
        {
            "event_timestamp": [
                pd.Timestamp(dt, unit="ms", tz="UTC").round("ms")
                for dt in pd.date_range(
                    start=datetime.now() - timedelta(days=10),
                    end=datetime.now(),
                    periods=1,
                )
            ],
            "driver_id": [1001],
        }
    )
print(entity_df)
print("Retrieving training data...")

training_df = fs.get_historical_features(features=features, entity_df=entity_df).to_df()

print()
print(training_df)
```
I use default parquet file which will be generated on init.

**Comment by Felix-neko** [comment]
> I use default parquet file which will be generated on init.

Thank you for you code example! And how did you upload the data from parquet to Hive base? I'll try to reproduce the case on my machine

**Comment by itsmesrds** [comment]
> > I use default parquet file which will be generated on init.
> 
> @itsmesrds , Thank you for you code example! And how did you upload the data from parquet to Hive base? I'll try to reproduce the case on my machine.
> 
> P.S. Also, how do you run Hive? Are you running it locally -- or you have access to some remote Hive cluster?

I took the reference from this readme (https://github.com/baineng/feast-hive#readme) to load the data to the cluster and here is the data file available (https://github.com/itsmesrds/file-upload/blob/main/driver_stats.parquet)

**Comment by Felix-neko** [comment]
Thanks! And what version of Hive and Hadoop do you use?

**Comment by itsmesrds** [comment]
> @itsmesrds Thanks! And what version of Hive and Hadoop do you use?

I'm using the sandbox that you shared and I'm able to access the data in hive cluster.

**Comment by baineng** [comment]
Thanks a lot for your work @Felix-neko. 
Sorry, I am very busy recently, will take time to review the changes.

- Could you format your code by 'make format' first?
- Have you passed all tests on your local environment?

**Comment by Felix-neko** [comment]
> * Could you format your code by 'make format' first?
Just done = )

> * Have you passed all tests on your local environment?
Yes, it passes the tests on my machine.

**Comment by Felix-neko** [comment]
> > @itsmesrds Thanks! And what version of Hive and Hadoop do you use?
> 
> I'm using the sandbox that you shared and I'm able to access the data in hive cluster. but feast is not able to read the data.

Oh! And what did you do to connect to hive-server explicitly? Did you connect via ssh?

P.S. When I was testing my code, I didn't upload my dataframe to DFS, I've uploaded it via Hive SQL interface (port 10000). I want to somehow try to upload parquet file but don't know how to get access to HDFS on my hive server yet = (

File: feast_hive/hive.py [hunk 1]
```diff
@@ -79,14 +90,22 @@ class HiveOfflineStoreConfig(FeastConfigBaseModel):
     kerberos_service_name: StrictStr = "impala"
     """ Specify particular impalad service principal. """
 
+    use_unique_field_names: StrictBool = False
```
Comment (baineng) [inline feast_hive/hive.py:93]: instead introduce a new variable here, why not just use `hive_conf` config?
`
hive_conf:
    hive.resultset.use.unique.column.names: false
`

**Comment by itsmesrds** [comment]
> P.S. When I was testing my code, I didn't upload my dataframe to HDFS as parquet file, I've uploaded it via Hive SQL interface (port 10000). I want to somehow try to upload parquet file but don't know how to get access to HDFS on my hive server yet = (

you can get it inside the hive-server container, you will be able to access the hdfs.  
```docker exec -it hive-server bash

you need to add the parquet file to Hive/employee path in your machine which is mounted in the sandbox of path /employee.

there you can run the below command, which will push the file to hdfs

hdfs dfs -copyFromLocal ./driver_stats.parquet /tmp/

Later create the hive table and load the data from hive shell as mentioned here https://github.com/baineng/feast-hive#create-hive-table

```

**Pull request merged**